allennlp/data/transforms.py

```diff
@@ -0,0 +1,531 @@
+from typing import Dict, Tuple, List, Iterable, Generic, TypeVar, Deque, Union
+import itertools
+from collections import deque, defaultdict
+import random
+
+import torch
+from torch.utils.data import IterableDataset as IterableTorchDataset
+
+
+from allennlp.common.registrable import Registrable
+from allennlp.data.instance import Instance
+from allennlp.data.vocabulary import Vocabulary
+
+from allennlp.data.fields import MetadataField
+
+from allennlp.common.util import lazy_groups_of
+
+
+Batched = Iterable[Instance]
```

Comment:
Nit: Why not `Batch`?

Reply:
We already have a `Batch` class which actually does some work, so I was trying to avoid confusion. I changed it to `Batch`.

Reply:
Oh wait, we already have a `Batch` transform in the same file. That's why this needed to be batched...